app/models/manageiq/providers/autosde/storage_manager.rb

```diff
@@ -296,4 +296,17 @@ def self.params_for_create
   def self.catalog_types
     {"autosde" => N_("Autosde")}
   end
+
+  def self.post_refresh_ems(ems_id, _)
+    events = EmsEvent.where("ems_id = ? AND (physical_storage_id IS NULL OR physical_storage_name IS NULL)", ems_id)
+
+    events.each do |e|
+      physical_storage_uuid = e.full_data[:storage_system]
+      physical_storage = PhysicalStorage.where(:ems_ref => physical_storage_uuid).to_a
```

Comment:
If `physical_storage_uuid` is unique within the scope of an EMS then what you want here is to use `find_by` instead of `.where.first` (the only reason the bot didn't warn is because you did `.to_a` then `[0]` below)
```suggestion
      physical_storage_uuid = e.full_data[:storage_system]
      physical_storage = PhysicalStorage.find_by(:ems_id => ems_id, :ems_ref => physical_storage_uuid)
```

Reply:
Changed